pyrobosim/pyrobosim/core/world.py

```diff
@@ -111,23 +111,39 @@ def set_inflation_radius(self, inflation_radius=0.0):
     ##########################
     # World Building Methods #
     ##########################
-    def add_room(self, room):
+    def add_room(self, **room_config):
         """
         Adds a room to the world.
 
         If the room does not have a specified name, it will be given an automatic
         name of the form ``"room0"``, ``"room1"``, etc.
 
-        If the room would cause a collision with another entity in the world,
+        If the room has an empty footprint or would cause a collision with another entity in the world,
         it will not be added to the world model.
 
-        :param room: Room object to add to the world.
-        :type room: :class:`pyrobosim.core.room.Room`
-        :return: True if the room was successfully added, else False.
-        :rtype: bool
+        :param \*\*room_config: Keyword arguments describing the room.
+
+            You can use ``room=Room(...)`` to directly pass in a :class:`pyrobosim.core.room.Room` object,
+            or alternatively use the same keyword arguments you would use to create a Room object.
+        :return: room object if successfully created, else None.
+        :rtype: :class:`pyrobosim.core.room.room`
         """
+
+        # If it's a room object, get it from the "room" named argument.
+        # Else, create a room directly from the specified arguments.
+        if "room" in room_config:
+            room = room_config["room"]
+        else:
+            room = Room(**room_config)
+
+        # If the room name is empty, automatically name it.
         if room.name is None:
-            room.name = f"room_{self.num_rooms}"
+            room.name = f"room{self.num_rooms}"
+
+        # If the room geometry is empty, do not allow it
+        if room.polygon.is_empty:
+            warnings.warn(f"Room {room.name} has empty geometry. Cannot add to world.")
+            return None
 
         # Check if the room collides with any other rooms or hallways
         is_valid_pose = True
```

Comment:
In the `for` loop here,  since we are interested only in the `is_valid_pose` changing state. 
We can avoid looping through all entities by breaking out of the loop the first time `is_valid_pose` becomes `False`

Reply:
Oh, nice catch! I'll add that in.